jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/resource/Resource.java

```diff
@@ -72,135 +67,41 @@ public abstract class Resource implements ResourceFactory
             return null;
         String scheme = uri.getScheme();
-        if (scheme == null)
-            return null;
-        if (!FileID.isArchive(uri))
+        if (scheme == null || !scheme.equalsIgnoreCase("jar"))
             return null;
         try
         {
-            if (scheme.equalsIgnoreCase("jar"))
-            {
-                return FileSystemPool.INSTANCE.mount(uri);
-            }
-            // TODO: review contract, should this be null, or an empty mount?
-            return null;
+            return FileSystemPool.INSTANCE.mount(uri);
         }
         catch (IOException ioe)
         {
             throw new IllegalArgumentException(ioe);
         }
     }
 
-    /**
-     * @param uri The URI to mount that requires a FileSystem (e.g. "jar:file://tmp/some.jar!/directory/file.txt")
-     * @return A reference counted {@link Mount} for that file system. Callers should call {@link Mount#close()} once
-     * they no longer require any resources from the mounted resource.
-     * @throws IOException If the uri could not be mounted.
-     * @throws IllegalArgumentException If the URI does not require a mount.
-     * @see #mountIfNeeded(URI)
-     */
-    public static Resource.Mount mount(URI uri) throws IOException
-    {
-        if (!FileID.isArchive(uri))
-            throw new IllegalArgumentException("URI is not a Java Archive: " + uri);
-        if (!uri.getScheme().equalsIgnoreCase("jar"))
-            throw new IllegalArgumentException("not an allowed URI: " + uri);
-        return FileSystemPool.INSTANCE.mount(uri);
-    }
-
-    /**
-     * @param path The path to a jar file to be mounted (e.g. "file:/tmp/some.jar")
-     * @return A reference counted {@link Mount} for that file system. Callers should call {@link Mount#close()} once
-     * they no longer require any resources from the mounted resource.
-     * @throws IOException If the path could not be mounted
-     */
-    public static Resource.Mount mountJar(Path path) throws IOException
-    {
-        if (!FileID.isArchive(path))
-            throw new IllegalArgumentException("Path is not a Java Archive: " + path);
-        URI pathUri = path.toUri();
-        if (!pathUri.getScheme().equalsIgnoreCase("file"))
-            throw new IllegalArgumentException("Not an allowed path: " + path);
-        URI jarUri = URIUtil.toJarFileUri(pathUri);
-        if (jarUri == null)
-            throw new IllegalArgumentException("Not a mountable archive: " + path);
-        return FileSystemPool.INSTANCE.mount(jarUri);
-    }
-
     /**
      * <p>Make a Resource containing a collection of other resources</p>
      * @param resources multiple resources to combine as a single resource. Typically, they are directories.
      * @return A Resource of multiple resources.
      * @see ResourceCollection
      */
-    public static ResourceCollection of(List<Resource> resources)
+    public static ResourceCollection combine(Collection<Resource> resources)
```

Comment:
List is important, as the search order is important within the ResourceCollection.
Make it only support `List<Resource>` please.

Reply:
Done.